pkg/controllers/provisioning/scheduling/inflightnode.go

```diff
@@ -87,9 +90,18 @@ func (n *InFlightNode) Add(pod *v1.Pod) error {
 		return err
 	}
 
-	if err := n.hostPortUsage.Add(pod); err != nil {
+	if err := n.hostPortUsage.Validate(pod); err != nil {
+		return err
+	}
+
+	// determine the number of volumes that will be mounted if the pod schedules
+	mountedVolumeCount, err := n.volumeUsage.Validate(pod)
```

Comment:
It's not clear to me why this can't be modeled as part of the node's capacity.

Reply:
It's not [reported correctly by the EBS CSI plugin](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/issues/1258) on the node and pods don't request volumes so it doesn't get reduced in the capacity.